jscomp/others/js_dict.mli

```diff
@@ -0,0 +1,64 @@
+(* Copyright (C) 2015-2016 Bloomberg Finance L.P.
+ * 
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * In addition to the permissions granted to you by the LGPL, you may combine
+ * or link a "work that uses the Library" with a publicly distributed version
+ * of this file to produce a combined library or application, then distribute
+ * that combined work under the terms of your choosing, with no requirement
+ * to comply with the obligations normally placed on you by section 4 of the
+ * LGPL version 3 (or the corresponding section of a later version of the LGPL
+ * should you choose to use a later version).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. *)
+
+
+type 'a t
+(** Dictionary type (ie an '\{ \}' JS object). However it is restricted 
+    to hold a single type; therefore values must have the same type. 
+    
+    This Dictionary type is mostly use with the [Js_json.t] type. *)
+
+type key = string
+(** Key type *)
+
+external get : 
+  'a t -> 
+  key -> 
+  'a option = "" [@@bs.get_index] [@@bs.return {undefined_to_opt}]
+(** [get dict key] returns [None] if the [key] is not found in the 
+    dictionary, [Some value] otherwise *)
+
+external unsafeGet : 'a t -> key -> 'a = "" [@@bs.get_index] 
+(** [unsafeGet dict key] return the value if the [key] exists, 
+    otherwise an {b undefined} value is returned. Must be used only 
+    when the existence of a key is certain. (i.e. when having called [keys]
+    function previously. 
+
+@example {[
+Array.iter (fun key -> Js.log (Js_dict.unsafeGet dic key)) (Js_dict.keys dict) 
+]} 
+*)
+
+val exnGet : 'a t -> key -> 'a
+(** [exnGet dict key] returns the value if the [key] exists, otherwise
```

Comment:
can we remove this API, this looks not very useful, exception based API is pervasive in OCaml mostly because it is efficient, but this characteristic is lost in JS land